tests/unit/testwebsockethandler.cpp

```diff
@@ -0,0 +1,207 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.orgß/MPL/2.0/. */
+
+#include "testwebsockethandler.h"
+#include "../../src/settingsholder.h"
+#include "../../src/constants.h"
+#include "../../src/mozillavpn.h"
+#include "helper.h"
+
+#include <QtWebSockets/QWebSocketServer>
+
+/**
+ * @brief Construct a new Mock Server:: Mock Server object
+ *
+ * Simple web socket server that exposes signals which tests can spy on.
+ */
+MockServer::MockServer() {
+  m_pWebSocketServer = new QWebSocketServer(
+      QStringLiteral("Mock Server"), QWebSocketServer::NonSecureMode, this);
+  if (m_pWebSocketServer->listen(QHostAddress::Any, 5000)) {
+    connect(m_pWebSocketServer, &QWebSocketServer::newConnection, this,
+            &MockServer::onNewConnection);
+  }
+}
+
+MockServer::~MockServer() {
+  m_pWebSocketServer->close();
+  qDeleteAll(m_clients.begin(), m_clients.end());
+}
+
+/**
+ * @brief Ackowledges a new client has connected to the server.
+ */
+void MockServer::onNewConnection() {
+  QWebSocket* pSocket = m_pWebSocketServer->nextPendingConnection();
+
+  emit newConnection(pSocket->request());
+
+  connect(pSocket, &QWebSocket::disconnected, this,
+          &MockServer::onSocketDisconnected);
+
+  m_clients << pSocket;
```

Comment:
This is a test, and I guess it's fine, but I would add this line before calling `emit newConnection`. Because maybe, in the future, we want to retrieve the list of clients in the slot, or we want to delete it immediately.